source/fundamentals/crud/write-operations/delete.txt

```diff
@@ -0,0 +1,115 @@
+
+The ``deleteOne()`` method deletes a single document. If the query
+filter matches more than one document, the method will remove the first
+occurrence of a match in the collection. 
+
+The ``deleteMany()`` method deletes all documents that match the query
+filter.
+
+The ``findOneAndDelete()`` method deletes the first occurrence of a
+match in the collection. 
+
+Examples
+--------
+
+The following examples are about a paint store that sells eight different
+colors of paint. The store had their annual online sale resulting in the
+following documents in their ``paint_inventory`` collection: 
+
+.. code-block:: json
+
+    { "_id": 1, "color": "red", "qty": 5 }
+    { "_id": 2, "color": "purple", "qty": 8 }
+    { "_id": 3, "color": "blue", "qty": 0 }
+    { "_id": 4, "color": "white", "qty": 0 }
+    { "_id": 5, "color": "yellow", "qty": 6 }
+    { "_id": 6, "color": "pink", "qty": 0 }
+    { "_id": 7, "color": "green", "qty": 0 }
+    { "_id": 8, "color": "black", "qty": 8 }
+
+``deleteMany()``:
+`````````````````
+
+The paint store website displays all documents in the
+``paint_inventory`` collection. To reduce customer confusion, the store
+wants to remove the colors that are out of stock (``qty`` is ``0``).
+
+To remove the out of stock colors, query the ``paint_inventory``
+collection where the ``qty`` is ``0`` and pass the query to the
+``deleteMany()`` method.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Delete.java
+   :language: java
+   :dedent:
+   :start-after: begin deleteManyExample
+   :end-before: end deleteManyExample
+
+The following shows the documents remaining in the ``paint_inventory``
+collection: 
+
+.. code-block:: json    
+
+    { "_id": 1, "color": "red", "qty": 5 }
+    { "_id": 2, "color": "purple", "qty": 8 }
+    { "_id": 5, "color": "yellow", "qty": 6 }
+    { "_id": 8, "color": "black", "qty": 8 }
+
+``deleteOne()``:
+````````````````
+
+The store is donating the remaining quantity of their red paint.
+This means that the ``red`` for red is now ``0`` and we need to
+remove red from the collection. 
+
+To remove red, query the ``paint_inventory`` collection where the
+``color`` is ``"red"`` and pass the query to the ``deleteOne()``
+method. 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Delete.java
+   :language: java
+   :dedent:
+   :start-after: begin deleteOneExample
+   :end-before: end deleteOneExample
+
+.. note::
+
+    In this scenario, you can also query all the documents in the
+    ``paint_inventory`` collection and pass the query to the
+    ``findOneAndDelete()`` method.  This works because red is the first
+    occurrence that matches the query filter. 
+
+    .. literalinclude:: /includes/fundamentals/code-snippets/Delete.java
+        :language: java
+        :dedent:
+        :start-after: begin findOneAndDeleteExample
+        :end-before: end findOneAndDeleteExample
```

Comment:
In my previous comment I meant we should remove the mention of ordering since it can be a deep subject. Maybe clarifying a bit and using "only" instead of "first"?